src/dryad/format/endf/createInterpolant.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef NJOY_DRYAD_FORMAT_ENDF_CREATEINTERPOLANT
+#define NJOY_DRYAD_FORMAT_ENDF_CREATEINTERPOLANT
+
+// system includes
+#include <vector>
+
+// other includes
+#include "tools/Log.hpp"
+#include "tools/std20/ranges.hpp"
+#include "dryad/type-aliases.hpp"
+
+namespace njoy {
+namespace dryad {
+namespace format {
+namespace endf {
+
+  /**
+   *  @brief Create an interpolation type from an ENDF interpolation type
+   */
+  InterpolationType createInterpolant( int interpolant ) {
+
+    switch ( interpolant ) {
+
+      case 1 : return InterpolationType::Histogram;
+      case 2 : return InterpolationType::LinearLinear;
+      case 3 : return InterpolationType::LinearLog;
+      case 4 : return InterpolationType::LogLinear;
+      case 5 : return InterpolationType::LogLog;
+      default : {
```

Comment:
Question, do you know what the 11-15 and 21-25 methods even are? The documentation is pretty bad for it.

Reply:
These are specifically for two-dimensional interpolation so they are not relevant for 1D functions like these. Unit base (21-25) means that you have to scale the domain of two 1D functions to [0,1] and then interpolate between them. If at one incident energy the spectrum is defined between 1-5 eV and for the next incident energy it is between 6-10 eV, you can actually interpolate on the spectrum using unit base.

For the corresponding points (11-15), I would have to check what exectly it is. Unit base is more common though.